In [58]:
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score
from sklearn.preprocessing import LabelEncoder

data = pd.read_csv("./cwfis_fwi2016.csv")

data.head()

aes     wmo             rep_date  temp   td    rh    ws    wg  wdir  \
0  1012475   71031  2016-01-01 12:00:00   2.9 -0.4  78.9  31.5   NaN    30   
1  1012710   71798  2016-01-01 12:00:00   2.9 -0.4  79.2  24.1  33.3    10   
2  1013998  719143  2016-01-01 12:00:00   4.1 -0.8  71.0  25.9   NaN    40   
3  1014820   71774  2016-01-01 12:00:00  -0.6 -2.9  84.6  15.5   NaN    13   
4  1016640   71778  2016-01-01 12:00:00   2.8 -0.3  80.3  40.7  50.0    40   

      pres     ...      sog  ffmc  dmc   dc  bui  isi  fwi  dsr  \
0  1028.80     ...      0.0  71.2  0.4  1.3  NaN  NaN  NaN  NaN   
1  1029.00     ...      0.0  79.0  0.9  2.9  NaN  NaN  NaN  NaN   
2  1029.29     ...      0.0   NaN  NaN  NaN  NaN  NaN  NaN  NaN   
3  1028.10     ...      0.0   NaN  NaN  NaN  NaN  NaN  NaN  NaN   
4  1028.70     ...      0.0  75.7  0.5  1.5  NaN  NaN  NaN  NaN   

                                       opts  calcstatus  
0                   IDW=RH:R:M=1:                     0  
1                            M=1:                     0  
2                 IDW=T:RH:R:M=1:                    -3  
3                    IDW=WSD:M=1:                    -3  
4                      IDW=R:M=1:                     0  

[5 rows x 23 columns]

In [59]:
df = data.dropna()

In [60]:
df.head()

aes    wmo             rep_date  temp   td    rh    ws    wg  wdir  \
20  1037553  71479  2016-01-01 12:00:00   8.2 -6.3  35.5  63.0  85.2   120   
26  1051351  71107  2016-01-01 12:00:00   7.6  4.0  78.2  46.3  57.4   160   
27  1054222  71472  2016-01-01 12:00:00   8.9  4.2  72.6  50.0  61.1   150   
30  1056869  71477  2016-01-01 12:00:00   6.0  2.6  79.0  57.4  70.4   130   
31  1057052  71111  2016-01-01 12:00:00   9.0  4.0  71.2  50.0  61.1   140   

      pres     ...      sog  ffmc  dmc    dc  bui   isi   fwi  dsr  \
20  1021.5     ...      0.0  84.9  1.1   3.9  1.3  49.8  19.2  5.1   
26  1020.2     ...      0.0  64.6  0.2   6.8  0.3   5.4   1.7  0.1   
27  1016.7     ...      0.0  81.4  1.7  10.6  2.5  16.5   9.3  1.4   
30  1019.8     ...      0.0  48.6  0.1   0.8  0.2   2.5   0.5  0.0   
31  1018.5     ...      0.0  78.9  0.6   5.4  1.0  12.7   5.9  0.6   

                                        opts  calcstatus  
20                            M=1:                     1  
26                            M=1:                     1  
27                            M=1:                     1  
30                            M=1:                     1  
31                            M=1:                     1  

[5 rows x 23 columns]

In [61]:
df['fire_percentage'] = (df['fwi']/30)*100

c:\users\john soowhan park\appdata\local\programs\python\python35\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [62]:
df.head()

aes    wmo             rep_date  temp   td    rh    ws    wg  wdir  \
20  1037553  71479  2016-01-01 12:00:00   8.2 -6.3  35.5  63.0  85.2   120   
26  1051351  71107  2016-01-01 12:00:00   7.6  4.0  78.2  46.3  57.4   160   
27  1054222  71472  2016-01-01 12:00:00   8.9  4.2  72.6  50.0  61.1   150   
30  1056869  71477  2016-01-01 12:00:00   6.0  2.6  79.0  57.4  70.4   130   
31  1057052  71111  2016-01-01 12:00:00   9.0  4.0  71.2  50.0  61.1   140   

      pres       ...         ffmc  dmc    dc  bui   isi   fwi  dsr  \
20  1021.5       ...         84.9  1.1   3.9  1.3  49.8  19.2  5.1   
26  1020.2       ...         64.6  0.2   6.8  0.3   5.4   1.7  0.1   
27  1016.7       ...         81.4  1.7  10.6  2.5  16.5   9.3  1.4   
30  1019.8       ...         48.6  0.1   0.8  0.2   2.5   0.5  0.0   
31  1018.5       ...         78.9  0.6   5.4  1.0  12.7   5.9  0.6   

                                        opts  calcstatus  fire_percentage  
20                            M=1:                     1        64.000000  
26                            M=1:                     1         5.666667  
27                            M=1:                     1        31.000000  
30                            M=1:                     1         1.666667  
31                            M=1:                     1        19.666667  

[5 rows x 24 columns]

In [63]:
df = df.drop(['aes', 'wmo', 'rep_date', 'td', 'wg', 'wdir', 'pres', 'vis', 'rndays', 'sog', 'ffmc', 'dmc', 'dc', 'bui', 'isi', 'dsr', 'opts', 'calcstatus'], axis=1)

In [64]:
df.head(20)

temp    rh    ws  precip   fwi  fire_percentage
20     8.2  35.5  63.0     0.0  19.2        64.000000
26     7.6  78.2  46.3     2.3   1.7         5.666667
27     8.9  72.6  50.0     0.0   9.3        31.000000
30     6.0  79.0  57.4     5.6   0.5         1.666667
31     9.0  71.2  50.0     0.0   5.9        19.666667
36     8.5  76.7  61.1     0.0   9.3        31.000000
38     6.2  64.8  55.6     0.0   9.8        32.666667
609   -1.0  80.3  22.2     0.0   0.3         1.000000
782    4.0  87.0  18.5     4.6   0.0         0.000000
852    5.9  82.4  50.0     0.0   8.3        27.666667
858    8.2  75.1  38.9     2.1   0.8         2.666667
859    8.8  85.1  44.4     0.0   6.7        22.333333
862    7.3  82.0  48.2     2.9   0.5         1.666667
863    9.0  76.3  27.8     0.0   1.9         6.333333
866    7.4  68.9  29.6     0.0   1.2         4.000000
1614   4.0  60.7  11.1     0.3   0.1         0.333333
1685   5.0  65.5   7.4     0.0   0.4         1.333333
1691   8.0  64.8  16.7     0.0   1.6         5.333333
1692   5.9  88.4  16.7     0.0   0.3         1.000000
1695   7.0  75.9  27.8     0.0   2.2         7.333333

In [65]:
df.shape

(52092, 6)

In [66]:
df.to_csv("fwidataset2016")

In [67]:
X = df.iloc[:,[0,1,2,3,]]

In [68]:
X.head()

temp    rh    ws  precip
20   8.2  35.5  63.0     0.0
26   7.6  78.2  46.3     2.3
27   8.9  72.6  50.0     0.0
30   6.0  79.0  57.4     5.6
31   9.0  71.2  50.0     0.0

In [69]:
y = df.fwi

In [70]:
y.head()

20    19.2
26     1.7
27     9.3
30     0.5
31     5.9
Name: fwi, dtype: float64

In [38]:
# X = pd.get_dummies(X, drop_first=True)

In [71]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.3)

In [72]:
X_train.head()

temp    rh    ws  precip
168236  15.0  77.2   9.3     0.0
192167  16.1  66.5  18.5     0.0
118910  10.2  91.1  18.5     1.4
165097  15.0  82.4  13.0     3.6
179440  15.0  87.9  14.8     9.4

In [73]:
linear = LinearRegression()
linear.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [74]:
pred = linear.predict(X_test)

In [75]:
r2_score(y_test,pred)

0.50688007445909289

* the result is not ideal as expected. Success rate for predicting FWI score based on tempearture, humidity, wind_speed, and rain precipitation is 51%.
* because our sensors will only have temp anyways and rely other data on other sources such as weather stations - which are not reliable - I will conduct linear regression prediction only using temp.